### PR TITLE
mvcc/reader: scan versions from u64::max to start_ts in get_txn… (#5062)

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
cherry-pick #5062

### CODE REVIEW

File: src/storage/mvcc/reader/mod.rs [hunk 1]
```diff
@@ -263,18 +263,15 @@ impl<S: Snapshot> MvccReader<S> {
...
                 return Ok(Some((commit_ts, write.write_type)));
             }
 
-            // If we reach a commit version whose type is not Rollback and start ts is
-            // larger than the given start ts, stop searching.
-            if write.write_type != WriteType::Rollback && write.start_ts > start_ts {
+            if commit_ts <= start_ts {
                 break;
             }
-
```
Comment (zhangjinpeng1987) [inline src/storage/mvcc/reader/mod.rs]: seek_ts = commit_ts - 1;

**Comment by youjiali1995** [comment]
/run-integration-cop-push-down-test tidb=release-2.1 pd=release-2.1 tidb-test=release-2.1

**Comment by youjiali1995** [comment]
/run-integration-cop-push-down-test tidb=release-2.1 pd=release-2.1

**Pull request merged**